x-pack/plugins/reporting/server/export_types/printable_pdf/create_job/index.ts

```diff
@@ -5,27 +5,41 @@
  * 2.0.
  */
 
+import { KibanaRequest } from 'src/core/server';
 import { cryptoFactory } from '../../../lib';
-import { CreateJobFn, CreateJobFnFactory } from '../../../types';
+import { CreateJobFn, CreateJobFnFactory, ReportingRequestHandlerContext } from '../../../types';
 import { validateUrls } from '../../common';
-import { JobParamsPDF, TaskPayloadPDF } from '../types';
+import { JobParamsPDF, JobParamsPDFLegacy, TaskPayloadPDF } from '../types';
+import { compatibilityShim } from './compatibility_shim';
 
+/*
+ * Incoming job params can be `JobParamsPDF` or `JobParamsPDFLegacy` depending
+ * on the version that the POST URL was copied from.
+ */
 export const createJobFnFactory: CreateJobFnFactory<
-  CreateJobFn<JobParamsPDF, TaskPayloadPDF>
+  CreateJobFn<JobParamsPDF | JobParamsPDFLegacy, TaskPayloadPDF>
 > = function createJobFactoryFn(reporting, logger) {
   const config = reporting.getConfig();
   const crypto = cryptoFactory(config.get('encryptionKey'));
 
-  return async function createJob(jobParams, _context, req) {
+  return compatibilityShim(async function createJobFn(
+    jobParams: JobParamsPDF,
+    _context: ReportingRequestHandlerContext,
+    req: KibanaRequest
+  ) {
     const serializedEncryptedHeaders = await crypto.encrypt(req.headers);
 
     validateUrls(jobParams.relativeUrls);
 
-    return {
+    const payload: TaskPayloadPDF = {
+      ...jobParams,
```

Comment:
Oops, this repeats the `relativeUrls` field of the job params (request) into the payload (saved report job)